src/OpenRasta/UriTemplateTable.cs

```diff
@@ -51,39 +51,44 @@ public void MakeReadOnly(bool allowDuplicateEquivalentUriTemplates)
       IsReadOnly = true;
       _keyValuePairsReadOnly = _keyValuePairs.AsReadOnly();
     }
-
     public Collection<UriTemplateMatch> Match(Uri uri)
     {
       var lastMaxLiteralSegmentCount = 0;
       var matches = new Collection<UriTemplateMatch>();
-      foreach (var template in KeyValuePairs)
+      if (BaseAddress != null)

```

Comment:
If I'm reading this right, if `BaseAddress` is `null` then no URL will ever find a match. Should `BaseAddress` be validated to not be `null` when initializing the `UriTemplateTable` instance?